### DISCUSSION TITLE
cannot open folder/files

### DISCUSSION THREAD

**Discussion by 20221027** [hneemann/Digital#1047] [Q&A]
i'm using a macbook pro M1 with parallels/win11 installed, java 1.8 is installed
on mac & win11 i can run the jar file (or exe in win11) & create a new project, at the end i save it
when i go back to the project, in win11 "digital' reloads the project because it was the last project i worked on
in mac if i want to open, all folders appear empty, which also happens in win11 if i want to open another project
therefor i only can create/save but not open an existing project because all folders appear empty (including the examples)
the 'new project' works, i can create a circuit & simulate it in both systems

i found that the monoflop needs a clock circuitry - why, the timing is normally set by a resistor/capacitor combo
please view 74123 & 4098 monoflops
in your simulator i would have to give the time(length) of the output pulse
thanks for reading

**Comment by hneemann** [comment]
see  #442

The simulation is purely event-based, there is no such thing as time.
Only if a simulation is operated with a clock of low frequency, a change of the clock input is triggered periodically. The simulation itself is then again exclusively event based.

**Comment by neivesten** [comment]
This is also happening to me on a M2. I wonder if that has anything to do with it. It works fine on native Windows 11. I'll have to try a VM for Windows on the MacBook. 

The problem is that folders appear empty, so trying to open a file is impossible. Trying to open a file by typing the path outputs an error: "Error Reading a file [file path] (Operation not permitted").

I haven't been able to find how to give permission, or what to give them to.

Thanks for reading.

**Reply by neivesten** [reply]
Update, found how to get it to work!

I was reading through the Pull Requests and found an user who was making some changes to improve the Mac experience.

You can read about it here https://github.com/hneemann/Digital/pull/750

Hneemann themselves uploaded a file that fixed the problem for me. It's one of the last messages. I'll update again if I encounter any more issues. So far so good.

Thanks Hneemann!
